x/cardchain/handler.go

```diff
@@ -49,6 +51,40 @@ func NewHandler(k keeper.Keeper) sdk.Handler {
 	}
 }
 
+func handleMsgReportMatch(ctx sdk.Context, keeper keeper.Keeper, msg *types.MsgReportMatch) (*sdk.Result, error) {
+	address, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, sdkerrors.Wrap(types.ErrInvalidAccAddress, "Invalid creator")
+	}
+
+	creator, err := keeper.GetUser(ctx, address)
+	if err != nil {
+		return nil, err
+	}
+	if creator.ReportMatches == false {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "Incorrect Reporter")
+	}
+
+	var matchId uint64
+	if mLen := len(keeper.GetAllMatches(ctx)); mLen == 0 {
```

Comment:
maybe this is not optimal, because this will get the iterator of all matches and then appends all matches and afterwards only reads out the length?

Reply:
I already changed this because it's retarded. You have to look at the summary of changes.